src/controllers/controller.doughnut.js

```diff
@@ -36,16 +36,25 @@ module.exports = function(Chart) {
 			labels: {
 				generateLabels: function(data) {
 					if (data.labels.length && data.datasets.length) {
+
 						return data.labels.map(function(label, i) {
+							var ds = data.datasets[0];
+							var arc = ds.metaData[i];
```

Comment:
@etimberg this change causes a bug when trying to add data to datasets, then calling chart.update(). `generateLabels()` is call before `buildOrUpdateElements()` (so before `addElements()`), meaning meta data are not yet initialized. You can test this issue with the `doughnut.html` example, and click *addData*.